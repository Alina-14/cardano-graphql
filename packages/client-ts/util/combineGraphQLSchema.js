const fs = require('fs');
const path = require('path');
const { loadFilesSync } = require('@graphql-tools/load-files');
const { mergeTypeDefs } = require('@graphql-tools/merge');
const { printSchemaWithDirectives } = require('@graphql-tools/utils');
const { makeExecutableSchema } = require('@graphql-tools/schema');

const cardanoDbHasuraSchema = loadFilesSync(path.resolve(__dirname, '..', 'api', 'cardano-db-hasura', 'schema.graphql'));
const genesisSchema = loadFilesSync(path.resolve(__dirname, '..', 'api', 'genesis', 'schema.graphql'));
const mergedTypes = mergeTypeDefs([cardanoDbHasuraSchema, genesisSchema], {
  throwOnConflict: true
});

const executableSchema = makeExecutableSchema({ typeDefs: [mergedTypes] });
const schema = printSchemaWithDirectives(executableSchema);

fs.writeFileSync(path.resolve(__dirname, '..','api', 'schema.graphql'), schema);
