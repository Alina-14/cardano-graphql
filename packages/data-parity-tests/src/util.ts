import fetch from 'node-fetch'

export async function getDataFromAPI (path: string) {
  const response = await fetch(`http://localhost:8201/${path}`)
  return response.json()
}
