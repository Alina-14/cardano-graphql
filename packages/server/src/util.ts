import { Application } from 'express'
import http from 'http'

export function listenPromise (app: Application, config: { port: number }): Promise<http.Server> {
  return new Promise(function (resolve, reject) {
    const server: http.Server = app.listen(config.port, () => resolve(server))
    server.on('error', reject)
  })
}
