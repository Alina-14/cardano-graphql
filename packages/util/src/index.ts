import { onFailedAttemptFor } from './onFailedAttemptFor'
import { loadQueryNode } from './queryNodeLoading'

export default {
  onFailedAttemptFor,
  loadQueryNode
}
