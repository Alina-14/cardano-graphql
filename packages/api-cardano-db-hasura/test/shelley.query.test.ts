import path from 'path'

import { DocumentNode } from 'graphql'
import util from '@cardano-graphql/util'
import { TestClient } from '@cardano-graphql/util-dev'
import { buildClient } from './util'

function loadQueryNode (name: string): Promise<DocumentNode> {
  return util.loadQueryNode(path.resolve(__dirname, 'queries'), name)
}

describe('Shelley era queries', () => {
  let mainnetClient: TestClient
  let mc4Client: TestClient
  beforeAll(async () => {
    mainnetClient = await buildClient('http://localhost:3100', 'http://localhost:8090', 5442)
    mc4Client = await buildClient('http://localhost:3102', 'http://localhost:8092', 5444)
  }, 15000)

  it('will not throw errors during the Byron era', async () => {
    const result = await mainnetClient.query({
      query: await loadQueryNode('shelleyEraQueries')
    })
    console.log(result)
    // expect(result.data).toMatchSnapshot()
  })
  it('Shelley era smoke test', async () => {
    const result = await mc4Client.query({
      query: await loadQueryNode('shelleyEraQueries')
    })
    console.log(result)
    // expect(result.data).toMatchSnapshot()
  })
})
