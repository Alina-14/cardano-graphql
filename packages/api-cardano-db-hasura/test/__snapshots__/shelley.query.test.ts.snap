// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Shelley era queries Shelley era smoke test 1`] = `
Object {
  "delegations": Array [
    Object {
      "address": "\\\\xe1e13c30ed2b3b721a8520b6948f547880c13b354fa09956b739b3c9c6",
      "stakePool": Object {
        "fixedCost": "340000000",
      },
    },
    Object {
      "address": "\\\\xe16799d296aded17f718a61651d2531c63aef9c13d3d38f432b9b3c5f2",
      "stakePool": Object {
        "fixedCost": "340000000",
      },
    },
    Object {
      "address": "\\\\xe145e1782462e2fe40aba7fd2314bf6dd5ed58954a922abe7a866ebeee",
      "stakePool": Object {
        "fixedCost": "340000000",
      },
    },
    Object {
      "address": "\\\\xe1c7e35526ed20029b2c3e2f58c8aa2b7aa728c90fc299367b99dbadda",
      "stakePool": Object {
        "fixedCost": "340000000",
      },
    },
    Object {
      "address": "\\\\xe1f7e3ac38a96e69d9ad7735a6f97c7ba7f8bbdc2c3758cd427cef1f2f",
      "stakePool": Object {
        "fixedCost": "340000000",
      },
    },
    Object {
      "address": "\\\\xe1c7e35526ed20029b2c3e2f58c8aa2b7aa728c90fc299367b99dbadda",
      "stakePool": Object {
        "fixedCost": "340000000",
      },
    },
    Object {
      "address": "\\\\xe1f7e3ac38a96e69d9ad7735a6f97c7ba7f8bbdc2c3758cd427cef1f2f",
      "stakePool": Object {
        "fixedCost": "340000000",
      },
    },
    Object {
      "address": "\\\\xe10a5b46ebc0e3dedabda03483fd72b2ab3d9417894cc53f9aebb214bd",
      "stakePool": Object {
        "fixedCost": "340000000",
      },
    },
    Object {
      "address": "\\\\xe10a5b46ebc0e3dedabda03483fd72b2ab3d9417894cc53f9aebb214bd",
      "stakePool": Object {
        "fixedCost": "340000000",
      },
    },
    Object {
      "address": "\\\\xe15afbb46e933d56024cd294668b07d0f3ff0f03ff94af3a646fe64907",
      "stakePool": Object {
        "fixedCost": "340000000",
      },
    },
  ],
  "delegations_aggregate": Object {
    "aggregate": Object {
      "avg": null,
      "count": "1759",
      "max": null,
      "sum": null,
    },
  },
  "stakeDeregistrations": Array [],
  "stakeDeregistrations_aggregate": Object {
    "aggregate": Object {
      "avg": null,
      "count": "0",
      "max": null,
      "sum": null,
    },
  },
  "stakePools": Array [
    Object {
      "fixedCost": "340000000",
      "hash": "069604bc8021b042866110e3990a9ed5f04671cc7fb915129c990fe1",
      "margin": 0.015,
      "metadataHash": "8c79c9df76ad1efd87800fa1fdcd23edfdb3062eaf645922689144a182454661",
      "owners": Array [],
      "pledge": "75000000000",
      "relays": Array [
        Object {
          "dnsName": null,
          "dnsSrvName": null,
          "ipv4": "15.165.202.246",
          "ipv6": null,
        },
        Object {
          "dnsName": null,
          "dnsSrvName": null,
          "ipv4": "3.126.65.17",
          "ipv6": null,
        },
        Object {
          "dnsName": null,
          "dnsSrvName": null,
          "ipv4": "3.20.137.144",
          "ipv6": null,
        },
      ],
      "retirement": null,
      "rewardAddress": "\\\\xe02917482e32082f9ce0d82723a467d9d37df78dc4ef43d41aa64f5467",
      "updatedIn": Object {
        "hash": "25170ff7a2978dc79de539565566b70c78d086695d0155054da7b50c49b044b9",
      },
      "url": "https://raw.githubusercontent.com/jlt03/9/master/poolmeta.json",
      "withdrawals": null,
    },
    Object {
      "fixedCost": "340000000",
      "hash": "fa1db746504016304258d7145cad6434562c5e63b14abe2edc6df016",
      "margin": 0.05,
      "metadataHash": "6e87e7c52f5b78d193f546e5e45f69bf5983c2a8c5ed62f50d8349ff72bd2291",
      "owners": Array [
        Object {
          "hash": "6f2bdee5fcc23e71f8f992d6e3779440226b714b35e0ba1d8ec5049f",
        },
      ],
      "pledge": "450000000",
      "relays": Array [
        Object {
          "dnsName": null,
          "dnsSrvName": null,
          "ipv4": "54.232.141.12",
          "ipv6": null,
        },
        Object {
          "dnsName": null,
          "dnsSrvName": null,
          "ipv4": "54.252.142.161",
          "ipv6": null,
        },
      ],
      "retirement": null,
      "rewardAddress": "\\\\xe01a8ec5ca20b2407264248152f5ef76dc36cebd57c4b7b10aec5191f9",
      "updatedIn": Object {
        "hash": "ac8dc7dc13df3520ab704090c37809968df919afb636e4555c2e4776057f946d",
      },
      "url": "https://moxietech.co/poolmeta.json",
      "withdrawals": null,
    },
    Object {
      "fixedCost": "340000000",
      "hash": "c2ab86326189c85b51055ca226aa6aecff3c41ad6baab988a92e535e",
      "margin": 0.03,
      "metadataHash": "f57f2b51042d5ac66127f347eb8b45e7cc0040d475437644c703b586d67c8520",
      "owners": Array [],
      "pledge": "500000000",
      "relays": Array [
        Object {
          "dnsName": null,
          "dnsSrvName": null,
          "ipv4": "34.249.11.89",
          "ipv6": null,
        },
      ],
      "retirement": null,
      "rewardAddress": "\\\\xe0a29d96efa58bfeb5dbf7cba19976697bc9a33987bc853f1d7b50bcce",
      "updatedIn": Object {
        "hash": "721b8496a35908bf3293c464c22a701ce4e48c3465b12be336b39364d7415af5",
      },
      "url": "https://bit.ly/3jLNIth",
      "withdrawals": null,
    },
    Object {
      "fixedCost": "340000000",
      "hash": "bf9278ddc16cf59f683759b4e9a3da38fe452d8dbb2523081348fd18",
      "margin": 0.009,
      "metadataHash": "64ba8d0ae2d2e50b910b243a1d938a603650065b2b63b311fd612d5bfa794ef5",
      "owners": Array [],
      "pledge": "25000000000",
      "relays": Array [
        Object {
          "dnsName": null,
          "dnsSrvName": null,
          "ipv4": null,
          "ipv6": null,
        },
      ],
      "retirement": null,
      "rewardAddress": "\\\\xe090b46c18e000525f0c6711330fc2c81ba5a370598b1d44069c7fbf7e",
      "updatedIn": Object {
        "hash": "d4f858d6c43375251f56ed86d8b407ad5be2e563efed18b7b805cfa0a28b880e",
      },
      "url": "https://viperstaking.com/assets/files/ADDER_metadata.json",
      "withdrawals": null,
    },
    Object {
      "fixedCost": "340000000",
      "hash": "3b44178eed1dce0cc5d45104f7bb248d16f0060d2f1b3461561d99c3",
      "margin": 0.06,
      "metadataHash": "95b9ea30932a4e7e908b040f6db0ee6127ad3df139a759358f1edb0ce48d4960",
      "owners": Array [],
      "pledge": "500000000",
      "relays": Array [
        Object {
          "dnsName": null,
          "dnsSrvName": null,
          "ipv4": "80.209.232.82",
          "ipv6": null,
        },
      ],
      "retirement": null,
      "rewardAddress": "\\\\xe065bc739246a362673cafeed10bde5e0009a95c9c5b99013d2b71ce08",
      "updatedIn": Object {
        "hash": "99a931f479dea44672597a43a9689e61a2911be4202cd2ba25a937b79572b99e",
      },
      "url": "https://timsahabutdinov.com/tims.json",
      "withdrawals": null,
    },
    Object {
      "fixedCost": "340000000",
      "hash": "9473c72c23c8239ca524101ca813a2d78f11c21dee45b6707d55f93c",
      "margin": 0.05,
      "metadataHash": "046c6c6fd17c509e823730e411f0328ba5249d5204b2e191d478f23237fe984f",
      "owners": Array [
        Object {
          "hash": "697ee41bc6bbcd05f93b5704f295d1f0ee3e243be96d76057817aeb9",
        },
      ],
      "pledge": "0",
      "relays": Array [
        Object {
          "dnsName": null,
          "dnsSrvName": null,
          "ipv4": null,
          "ipv6": null,
        },
      ],
      "retirement": null,
      "rewardAddress": "\\\\xe0804530b5c7ea2326481d6f3d05bdc224b9a26b87355a5c36baf48851",
      "updatedIn": Object {
        "hash": "4436d9ba194c3a56ec023310e61424ef25fb7b467449ddd395293cc2a3412ae0",
      },
      "url": "https://tokyostaker.com/metadata/jp2.json",
      "withdrawals": null,
    },
    Object {
      "fixedCost": "340000000",
      "hash": "3552559c39774dc5c65d3c2c7c6f447b0c5abaa54c7d7ea0b31987d1",
      "margin": 0.06,
      "metadataHash": "f9d5e977d38c1eb387cfdec0677284d11a3f9129855b542bd332d636f9bd7bbf",
      "owners": Array [],
      "pledge": "500000000000",
      "relays": Array [
        Object {
          "dnsName": null,
          "dnsSrvName": null,
          "ipv4": null,
          "ipv6": null,
        },
        Object {
          "dnsName": null,
          "dnsSrvName": null,
          "ipv4": null,
          "ipv6": null,
        },
      ],
      "retirement": null,
      "rewardAddress": "\\\\xe098bfe94f29010ce9841726c10f6907878094ce857dc53a3cf3f426e9",
      "updatedIn": Object {
        "hash": "4eb78bf81f431090f5f80d94fe6b9b704299c627c6c748d579894a6661ddaa86",
      },
      "url": "https://tinyurl.com/y3aryqrz",
      "withdrawals": null,
    },
    Object {
      "fixedCost": "340000000",
      "hash": "da67c9a99b415fa6fd6896d9f46153e5429627c92da626d86bab530a",
      "margin": 0.01,
      "metadataHash": "08db1e0079c47a845d48e0093f4e71c593839f3f9af4ce1903859960d0af4c4f",
      "owners": Array [
        Object {
          "hash": "56cbf8a007ba1e99bfc34410051cd1f68ed06b6de6e3fd3cd29b70d7",
        },
      ],
      "pledge": "13597251792",
      "relays": Array [
        Object {
          "dnsName": null,
          "dnsSrvName": null,
          "ipv4": "142.93.179.107",
          "ipv6": null,
        },
      ],
      "retirement": null,
      "rewardAddress": "\\\\xe0d36b2e1ee57867071015e7d8cd8de8fa7a73224a5f0303dde5c55d22",
      "updatedIn": Object {
        "hash": "6435fd151ed52b309b8217841660c926b5784c46686a4530d51ef7ccb1f8f7d7",
      },
      "url": "https://mentatpool.com/mntat_meta.json",
      "withdrawals": null,
    },
    Object {
      "fixedCost": "340000000",
      "hash": "dbc3ec57ad1f05dd0c3620cfbb37d0aca3003bcbed0c77d7bd42fc43",
      "margin": 0.05,
      "metadataHash": "6bdf778e40bff27fe8e5e35fe3a52f7d13baea40d083e665e09472c391b2cd3b",
      "owners": Array [
        Object {
          "hash": "2ff25aa7fcd4cfd5b2047828104bf104e7f03eb9297f7751746a3459",
        },
      ],
      "pledge": "1000000000",
      "relays": Array [
        Object {
          "dnsName": null,
          "dnsSrvName": null,
          "ipv4": null,
          "ipv6": null,
        },
        Object {
          "dnsName": null,
          "dnsSrvName": null,
          "ipv4": null,
          "ipv6": null,
        },
        Object {
          "dnsName": null,
          "dnsSrvName": null,
          "ipv4": null,
          "ipv6": null,
        },
        Object {
          "dnsName": null,
          "dnsSrvName": null,
          "ipv4": null,
          "ipv6": null,
        },
        Object {
          "dnsName": null,
          "dnsSrvName": null,
          "ipv4": null,
          "ipv6": null,
        },
        Object {
          "dnsName": null,
          "dnsSrvName": null,
          "ipv4": null,
          "ipv6": null,
        },
        Object {
          "dnsName": null,
          "dnsSrvName": null,
          "ipv4": null,
          "ipv6": null,
        },
        Object {
          "dnsName": null,
          "dnsSrvName": null,
          "ipv4": null,
          "ipv6": null,
        },
        Object {
          "dnsName": null,
          "dnsSrvName": null,
          "ipv4": null,
          "ipv6": null,
        },
        Object {
          "dnsName": null,
          "dnsSrvName": null,
          "ipv4": null,
          "ipv6": null,
        },
        Object {
          "dnsName": null,
          "dnsSrvName": null,
          "ipv4": null,
          "ipv6": null,
        },
      ],
      "retirement": null,
      "rewardAddress": "\\\\xe078d311b645987717f85e619158d4689e24cc11117e9470a20418a9fc",
      "updatedIn": Object {
        "hash": "d20df3be555c172a5a8aa433fde1bc14e8186abde81a180bddda6452c811c9b9",
      },
      "url": "https://www.titanstaking.io/titan.metadata.json",
      "withdrawals": null,
    },
    Object {
      "fixedCost": "340000000",
      "hash": "7b37465168b0c53bfa98ec3c093def47ffb10062480d5b009c9f7b17",
      "margin": 0.25,
      "metadataHash": "49f4d3ebcf68bfd29412706e78f2ad6e5e11453c826d60f0432be99ad6f5200a",
      "owners": Array [],
      "pledge": "100000000000",
      "relays": Array [
        Object {
          "dnsName": null,
          "dnsSrvName": null,
          "ipv4": "78.47.166.84",
          "ipv6": null,
        },
      ],
      "retirement": null,
      "rewardAddress": "\\\\xe0b25f9d5aec583a012240e4c07098403b5bd0cabc927320ef15cc5a64",
      "updatedIn": Object {
        "hash": "868c942e6f0165ff7902b88645101dfa11d02d9d0ca76fd5d5c4ffdd224c65b3",
      },
      "url": "https://raw.githubusercontent.com/truststakepool/a/master/a.json",
      "withdrawals": null,
    },
  ],
  "stakePools_aggregate": Object {
    "aggregate": Object {
      "avg": Object {
        "margin": 0.0413847848360656,
      },
      "count": "488",
      "max": Object {
        "margin": 1,
      },
      "sum": Object {
        "margin": 20.195775,
      },
    },
  },
  "stakeRegistrations": Array [
    Object {
      "address": "\\\\xe1e13c30ed2b3b721a8520b6948f547880c13b354fa09956b739b3c9c6",
      "transaction": Object {
        "hash": "01302f3caec79e9e1d9088e481b4b95bc581c936f0adf839c9d9490554530aaa",
      },
    },
    Object {
      "address": "\\\\xe10a5b46ebc0e3dedabda03483fd72b2ab3d9417894cc53f9aebb214bd",
      "transaction": Object {
        "hash": "6783eee716eaa88f6b859522d687363f3e20c47d8c6928a25edec2a8a25abad4",
      },
    },
    Object {
      "address": "\\\\xe16799d296aded17f718a61651d2531c63aef9c13d3d38f432b9b3c5f2",
      "transaction": Object {
        "hash": "fd5bb0da9fdcb0c9bb71540b531d900e07e9e1da609213bdbeec033ed0ad0387",
      },
    },
    Object {
      "address": "\\\\xe145e1782462e2fe40aba7fd2314bf6dd5ed58954a922abe7a866ebeee",
      "transaction": Object {
        "hash": "536283d6f4b54c02b8640d2f59fa97281cd11dcc69b345cf85a16f49b3a289bb",
      },
    },
    Object {
      "address": "\\\\xe1f7e3ac38a96e69d9ad7735a6f97c7ba7f8bbdc2c3758cd427cef1f2f",
      "transaction": Object {
        "hash": "21a3204b928e52ceca2ba10460fac3c5eae8af8a7a09f674d87cf6a93d99b1f9",
      },
    },
    Object {
      "address": "\\\\xe1c7e35526ed20029b2c3e2f58c8aa2b7aa728c90fc299367b99dbadda",
      "transaction": Object {
        "hash": "760bed55c23f21d650f107573564f1652949c64561d89ed393777be838166cf8",
      },
    },
    Object {
      "address": "\\\\xe15afbb46e933d56024cd294668b07d0f3ff0f03ff94af3a646fe64907",
      "transaction": Object {
        "hash": "0ca4ade051eb1484fd3fdfffcf5a53fea95a68b5075e1ba3e3fb57cb7b3b074b",
      },
    },
    Object {
      "address": "\\\\xe1ea94837cb0ed7051b9e3584d95c0360e1ee731dace81f177c57b8024",
      "transaction": Object {
        "hash": "41657a1c7608bc60e2958ba5e0d59931a5fefcb5544b3eb618f62c2c5ead3a50",
      },
    },
    Object {
      "address": "\\\\xe16ec91081410b8ca5f2436dbeedf5f86a25fb2816cff29f5693a57081",
      "transaction": Object {
        "hash": "d9f1798d790fb6bd999ba8fdb65a080a6416f5422a1fc30758082ecbf37b640e",
      },
    },
    Object {
      "address": "\\\\xe103af0bab010160adba8947def6de4203520bf75bc58210a4089498ee",
      "transaction": Object {
        "hash": "62dad4a28057f208bc2ade8676015131809782590b91b8fcb88959f7edff9d26",
      },
    },
  ],
  "stakeRegistrations_aggregate": Object {
    "aggregate": Object {
      "avg": null,
      "count": "23862",
      "max": null,
      "sum": null,
    },
  },
  "withdrawals": Array [
    Object {
      "address": "\\\\xe1777c9b83eb57b1b9f838132089c34dca6d2c8c556d1f6807664f6b51",
      "amount": "84700372786",
      "transaction": Object {
        "hash": "abfa3a40f4b6893c3a091b853945803c56c53dd5730e2535b314cdde6a4644ba",
      },
    },
    Object {
      "address": "\\\\xe164a78878a5b12455cd20a64a6052b1fe5466f6617b253a2083b7bd5f",
      "amount": "284886079252",
      "transaction": Object {
        "hash": "96853bba40d047fd948a763f3fee2941647274a9a4cb73aff566aca1df7e3fed",
      },
    },
    Object {
      "address": "\\\\xe125df996af19ebaf5e863b79265c03f81c8dcfd82d88a7072816d64b4",
      "amount": "17614173866",
      "transaction": Object {
        "hash": "a4adcdf7fd3e184985e2eeafbb6539b781ab8fc95afd3dda73514ec100b71dd1",
      },
    },
    Object {
      "address": "\\\\xe1ad6564703067fea17dd03549688cae938e685954a3a5c1b1db336f2a",
      "amount": "292995411146",
      "transaction": Object {
        "hash": "cc0b8300b51329b699cf70e8a7dbb898f1ab54faa7616f3aba65010b1e855af9",
      },
    },
    Object {
      "address": "\\\\xe14f88502399c5d8a3aa89e7ae8e126a2bac1ff88c2d2843c095396d5e",
      "amount": "48192857486",
      "transaction": Object {
        "hash": "b24dcdf303253201ac3ec1620c14323d15083e2455c88beae944449c3276eacb",
      },
    },
    Object {
      "address": "\\\\xe1b46b5c5e12580a0447e309aca800e7e72aa02ab28a0087cfb3643df5",
      "amount": "14073914721",
      "transaction": Object {
        "hash": "dc2096c4faa0019ef282b18c8aa19da613a78a7b67116d360e0af4c1c2bb2fcf",
      },
    },
    Object {
      "address": "\\\\xe199443aae8e5b7b304405d461db9bfd19923c9f547e017a87c17fd2f0",
      "amount": "16102066702",
      "transaction": Object {
        "hash": "d8d126f7d1924925f3b5b63a156732448ca4a0c0e67ca62fc4280f0b8af6cbd2",
      },
    },
    Object {
      "address": "\\\\xe128ecd66a3be4853fd02b5934821dc7bd81a3c429472c82b02760fce2",
      "amount": "26815866249",
      "transaction": Object {
        "hash": "af9cdfc65fbc05e39d8403b25eb3dd23c71ab638e23d6bcea6c88000e329a71a",
      },
    },
    Object {
      "address": "\\\\xe1b48e1d28ae9d4ea604ec265551d177cd2b5ccb18818c7f1b70cfd42a",
      "amount": "50869832914",
      "transaction": Object {
        "hash": "3c62f171e766adb883f07124ef386f629f7826fa85a9febb86f5b79fd549e37d",
      },
    },
    Object {
      "address": "\\\\xe17e5eff8ebf9861f3841f45b4661703dd5afd188d58cec6c21b848d5a",
      "amount": "500000000",
      "transaction": Object {
        "hash": "c1b7388c8502b17fed36ae6004076d440b7b62379c1ac78f3ddf5a8f41678558",
      },
    },
  ],
}
`;

exports[`Shelley era queries will not throw errors during the Byron era 1`] = `
Object {
  "delegations": Array [],
  "delegations_aggregate": Object {
    "aggregate": Object {
      "avg": null,
      "count": "0",
      "max": null,
      "sum": null,
    },
  },
  "stakeDeregistrations": Array [],
  "stakeDeregistrations_aggregate": Object {
    "aggregate": Object {
      "avg": null,
      "count": "0",
      "max": null,
      "sum": null,
    },
  },
  "stakePools": Array [],
  "stakePools_aggregate": Object {
    "aggregate": Object {
      "avg": Object {
        "margin": null,
      },
      "count": "0",
      "max": Object {
        "margin": null,
      },
      "sum": Object {
        "margin": null,
      },
    },
  },
  "stakeRegistrations": Array [],
  "stakeRegistrations_aggregate": Object {
    "aggregate": Object {
      "avg": null,
      "count": "0",
      "max": null,
      "sum": null,
    },
  },
  "withdrawals": Array [],
}
`;
